internal/actions/update.go

```diff
@@ -123,11 +127,58 @@ func checkDependencies(containers []container.Container) {
 	}
 }
 
+func executePreCheck(client container.Client, params UpdateParams) {
+	containers, err := client.ListContainers(params.Filter)
+	if err != nil {
+		return
+	}
+	for _, container := range containers {
+		executePreCheckCommand(client, container)
+	}
+}
+
+func executePostCheck(client container.Client, params UpdateParams) {
+	containers, err := client.ListContainers(params.Filter)
+	if err != nil {
+		return
+	}
+	for _, container := range containers {
+		executePostCheckCommand(client, container)
+	}
+}
+
+func executePreCheckCommand(client container.Client, container container.Container) {
```

Comment:
Should we maybe extract this logic to a shared method doing all the work, taking the step and the command as a argument? Ie. `executePostCheck(client container.Client, params UpdateParams)` calling `executeCommandInContainer(client container.Client, params UpdateParams, command string, step string)? This includes `executePreUpdate` and `executePostUpdate` as well.